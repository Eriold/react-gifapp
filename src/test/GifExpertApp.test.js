import React from 'react'
import GifExpertApp from '../GifExpertApp'
import { shallow } from 'enzyme'

describe('render <GifExpertApp />', () => {
  test('show <GifExpertApp /> ', () => {
    const wrapper = shallow(<GifExpertApp />);
    expect(wrapper).toMatchSnapshot();
  })

  test('show cant list category', () => {
    const categories = ['One Punch', 'Overlord'];
    const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('GifGrid').length).toBe(categories.length)
  })

}) 