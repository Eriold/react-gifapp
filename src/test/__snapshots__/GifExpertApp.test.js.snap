// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render <GifExpertApp /> show <GifExpertApp />  1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`render <GifExpertApp /> show cant list category 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Overlord"
      key="Overlord"
    />
  </ol>
</Fragment>
`;
